### PR TITLE
Fix "Could not find config file" error when creating config in already existing directory

### PR DESCRIPTION
**Pull request by anonymous123-code**
Makes it so that the check wether a new config should be created in `new_with_data_dir` also checks if the config file exists, not only the directory.
Also adds corresponding test cases.

### CODE REVIEW

**Review by Oliver-makes-code — APPROVED** [review]
lgtm, thanks for adding some tests too

File: helixlauncher-core/Cargo.toml [hunk 1]
```diff
@@ -32,8 +32,8 @@ anyhow = "1.0.69" # FIXME: remove
...
 tempfile-fast = "0.3.4"
 
-[build-dependencies]
-cbindgen = { version = "0.24", optional = true }
-
 [dev-dependencies]
 tempfile = "3.6.0"
+
+[build-dependencies]
+cbindgen = { version = "0.24", optional = true }
```
Comment (kb-1000) [inline helixlauncher-core/Cargo.toml:39]: why are you reordering them?

File: helixlauncher-core/src/config.rs [hunk 2]
```diff
@@ -165,3 +165,24 @@ fn get_base_path() -> PathBuf {
         .or_else(|| env::current_dir().ok())
         .unwrap()
 }
+
+#[cfg(test)]
+mod tests {
+    use anyhow::{Ok, Result};
```
Comment (kb-1000) [inline helixlauncher-core/src/config.rs:171]: Same question here, does the builtin `Ok` not suffice?

File: helixlauncher-core/src/config.rs [hunk 3]
```diff
@@ -165,3 +165,24 @@ fn get_base_path() -> PathBuf {
...
+
+    use crate::config::Config;
+
+    #[tokio::test]
+    async fn create_config_in_non_existing_dir() -> Result<()> {
+        let dir = tempfile::tempdir()?;
+        let _ = Config::new_with_data_dir("", "", dir.path().join("abc"))?;
```
Comment (kb-1000) [inline helixlauncher-core/src/config.rs:178]: As these parameters are present, they should get proper values. (tbh I don't fully remember why I left them in even for `new_with_data_dir`, but I'm sure I had reasons)

**Review by kb-1000 — CHANGES_REQUESTED** [review]

**Review by kb-1000 — APPROVED** [review]

**Pull request merged**